ipa-core/src/protocol/ipa_prf/mod.rs

```diff
@@ -225,4 +262,61 @@ pub mod tests {
             );
         });
     }
+
+    #[test]
+    fn test_shuffle_inputs() {
+        run(|| async {
+            let world = TestWorld::default();
+
+            let records: Vec<TestRawDataRecord> = vec![
+                TestRawDataRecord {
+                    timestamp: 0,
+                    user_id: 12345,
+                    is_trigger_report: false,
+                    breakdown_key: 1,
+                    trigger_value: 0,
+                },
+                TestRawDataRecord {
+                    timestamp: 0,
+                    user_id: 12345,
+                    is_trigger_report: false,
+                    breakdown_key: 2,
+                    trigger_value: 0,
+                },
+                TestRawDataRecord {
+                    timestamp: 10,
+                    user_id: 12345,
+                    is_trigger_report: true,
+                    breakdown_key: 0,
+                    trigger_value: 5,
+                },
+                TestRawDataRecord {
+                    timestamp: 0,
+                    user_id: 68362,
+                    is_trigger_report: false,
+                    breakdown_key: 1,
+                    trigger_value: 0,
+                },
+                TestRawDataRecord {
+                    timestamp: 20,
+                    user_id: 68362,
+                    is_trigger_report: true,
+                    breakdown_key: 0,
+                    trigger_value: 2,
+                },
+            ];
+
+            let mut result: Vec<TestRawDataRecord> = world
+                .semi_honest(records.clone().into_iter(), |ctx, input_rows| async move {
+                    shuffle_inputs::<_, BA8, BA3, BA20>(ctx, input_rows)
+                        .await
+                        .unwrap()
+                })
+                .await
+                .reconstruct();
```

Comment:
Compiler unable to compile this reconstruct 
"the method `reconstruct` exists for array `[Vec<OprfReport<BA8, BA3, BA20>>; 3]`, but its trait bounds were not satisfied
items from traits can only be used if the trait is implemented and in scoperustc
(rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2FUsers%2Frichaj%2FProposals%2FIPA%2Fraw-ipa%2Fipa-core%2Fsrc%2Fprotocol%2Fipa_prf%2Fmod.rs)
mod.rs(309, 54):
sharing.rs(82, 27): trait bound `[&'v [OprfReport<boolean_array_8::BA8, boolean_array_3::BA3, boolean_array_20::BA20>]; 3]: test_fixture::sharing::Reconstruct<Vec<_>>` was not satisfied"

Reply:
done. please note that we need to increase the size of the input in order to assert random order of shuffle

Reply:
thanks a ton as always :)